### PR TITLE
Add make commands for deploying on test cluster

### PR DESCRIPTION
**Pull request by vipin-dfe**
### Context

Enable the RSM application to deploy on Kubernetes cluster.

### Changes proposed in this pull request
The PR adds Makefile commands to allow the RSM application to be deployed to a Kubernetes cluster. 

### Guidance to review
The following files have been updated 
- `Makefile` has been updated
-  Configuration added to `terraform/application/config`

### Link to Trello card
https://trello.com/c/TndXQw6F

### Checklist

- [ ] Attach to Trello card
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally
- [ ]

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -181,6 +181,15 @@ aks-review: test-cluster
...
 	$(eval include global_config/review.sh)
 
+.PHONY: aks-test
+aks-test: test-cluster
+	$(eval DEPLOY_ENV=test)
+	$(eval ENV_SHORT=ts)
+	$(eval ENV_TAG=test)
+	$(eval NAME_ENV=${DEPLOY_ENV})
+	$(eval RESOURCE_ENV=${ENV_SHORT})
+	$(eval include global_config/test.sh)
```
Comment (saliceti) [inline Makefile:191]: global_config/test.sh was not added to the PR

File: Makefile [hunk 2]
```diff
@@ -181,6 +181,15 @@ aks-review: test-cluster
 	$(eval ENVIRONMENT=pr-${PR_NUMBER})
 	$(eval include global_config/review.sh)
 
+.PHONY: aks-test
+aks-test: test-cluster
+	$(eval DEPLOY_ENV=test)
```
Comment (saliceti) [inline Makefile:186]: DEPLOY_ENV, ENV_SHORT, ENV_TAG, NAME_ENV are not used in the new aks-* targets

File: terraform/application/config/test.tfvars.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "cluster": "test",
+  "namespace": "tra-development",
+  "deploy_azure_backing_services": false,
+  "enable_postgres_ssl": false,
+  "webapp_startup_command": [
```
Comment (saliceti) [inline terraform/application/config/test.tfvars.json:6]: The default command from the Dockerfile should be used by default so there is no need to override

File: terraform/application/config/test.tfvars.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "cluster": "test",
+  "namespace": "tra-development",
+  "deploy_azure_backing_services": false,
```
Comment (saliceti) [inline terraform/application/config/test.tfvars.json:4]: The test environment uses real databases. deploy_azure_backing_services and enable_postgres_ssl should not be set here, so the default values are used

File: terraform/application/config/test.tfvars.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{
+  "cluster": "test",
+  "namespace": "tra-development",
```
Comment (saliceti) [inline terraform/application/config/test.tfvars.json:3]: According to [the onboarding form](https://hedgedoc.teacherservices.cloud/eM_-E7fHTJWbxNKEIpYwTw?view#Onboarding-form-data) test should be deployed to the tra-test namespace

**Comment by vipin-dfe** [comment]
[RSM: build test environment](https://trello.com/c/TndXQw6F/2101-rsm-build-test-environment)

File: Makefile (hunk 1 above)
Comment (vipin-dfe) [reply Makefile:187]: test.sh added.

File: Makefile (hunk 2 above)
Comment (vipin-dfe) [reply Makefile]: Remove additional variables.

File: terraform/application/config/test.tfvars.json (hunk 4 above)
Comment (vipin-dfe) [reply terraform/application/config/test.tfvars.json]: comment applied.

File: terraform/application/config/test.tfvars.json (hunk 5 above)
Comment (vipin-dfe) [reply terraform/application/config/test.tfvars.json]: comment applied.

File: global_config/test.sh [hunk 6]
```diff
@@ -0,0 +1,7 @@
+CONFIG=test
+ENVIRONMENT=test
+CONFIG_SHORT=ts
+AZURE_SUBSCRIPTION=s189-teacher-services-cloud-test
+AZURE_RESOURCE_PREFIX=s189t01
+KV_PURGE_PROTECTION=false
+RESOURCE_NAME_PREFIX=s189t01
```
Comment (saliceti) [inline global_config/test.sh:7]: I believe RESOURCE_NAME_PREFIX is not use in aks

File: Makefile [hunk 7]
```diff
@@ -181,6 +181,11 @@ aks-review: test-cluster
 	$(eval ENVIRONMENT=pr-${PR_NUMBER})
 	$(eval include global_config/review.sh)
 
+.PHONY: aks-test
+aks-test: test-cluster
+	$(eval RESOURCE_ENV=${ENV_SHORT})
```
Comment (saliceti) [inline Makefile:186]: I believe RESOURCE_ENV is not used in AKS

File: terraform/application/config/test_Terrafile [hunk 8]
```diff
@@ -0,0 +1,3 @@
+aks:
+    source:  "https://github.com/DFE-Digital/terraform-modules"
+    version: "main"
```
Comment (saliceti) [inline terraform/application/config/test_Terrafile:3]: This should the testing tag as it's more stable: https://github.com/DFE-Digital/terraform-modules/?tab=readme-ov-file#testing-tag

Comment (vipin-dfe) [reply terraform/application/config/test_Terrafile:3]: done.

File: global_config/test.sh (hunk 6 above)
Comment (vipin-dfe) [reply global_config/test.sh]: done.

File: Makefile (hunk 7 above)
Comment (vipin-dfe) [reply Makefile]: done.

File: terraform/application/config/test_Terrafile (hunk 8 above)
Comment (vipin-dfe) [reply terraform/application/config/test_Terrafile]: done.

File: terraform/application/config/test.tfvars.json (hunk 3 above)
Comment (vipin-dfe) [reply terraform/application/config/test.tfvars.json]: done.

File: .github/workflows/build-and-deploy.yml [hunk 9]
```diff
@@ -106,6 +106,31 @@ jobs:
...
+  deploy_aks:
+    name: Deploy to ${{ matrix.environment }} environment for AKS
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'
+    concurrency: deploy_${{ matrix.environment }}
+    needs: [build_image]
+    strategy:
+      max-parallel: 1
+      matrix:
+        environment: [aks-test, aks-preprod]
```
Comment (saliceti) [inline .github/workflows/build-and-deploy.yml:118]: aks-preprod is not ready, should we remove it from this list for now?

Comment (vipin-dfe) [reply .github/workflows/build-and-deploy.yml]: removed.

File: .github/workflows/build-and-deploy.yml [hunk 10]
```diff
@@ -106,6 +106,31 @@ jobs:
             | ----------------- | ------------------------------------------------------------------------------------------------------------------------------------------|
             | Refer Serious Misconduct      | <https://refer-serious-misconduct-pr-${{ github.event.pull_request.number }}.test.teacherservices.cloud>            |
 
+  deploy_aks:
+    name: Deploy to ${{ matrix.environment }} environment for AKS
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'
```
Comment (saliceti) [inline .github/workflows/build-and-deploy.yml:112]: This should only run on push to main. Can we add to the condition?

Comment (vipin-dfe) [reply .github/workflows/build-and-deploy.yml]: done.

File: terraform/application/database.tf [hunk 11]
```diff
@@ -8,6 +8,7 @@ module "postgres" {
   service_short               = var.service_short
   config_short                = var.config_short
   cluster_configuration_map   = module.cluster_data.configuration_map
+  azure_extensions            = ["PGCRYPTO"]
```
Comment (shaheislamdfe) [inline terraform/application/database.tf:11]: I don't recall seeing this value before, what does it do?

**Review by shaheislamdfe — APPROVED** [review]

**Pull request merged**